### PR TITLE
feat(connector): validate sink primary key and sink type on connector node

### PR DESCRIPTION
**Pull request by xx01cyx**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

For JDBC sink:
- If the JDBC table has no pk, we reject the upsert sink.
- If the JDBC table does have pk, we reject the upsert sink if the user has defined pk.
- Otherwise, sink validation passes.

Also validate columns for JDBC sink.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by tabVersion — APPROVED** [review]
LGTM good work!

**Review by wenym1 — APPROVED** [review]

**Review by wenym1 — COMMENTED** [review]
any reason to disablr the jdbc sink e2e test?

**Comment by xx01cyx** [comment]
> any reason to disablr the jdbc sink e2e test?

Will enable the test in #8610 after this PR gets merged.

File: java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSinkFactory.java [hunk 1]
```diff
@@ -47,12 +53,56 @@ public void validate(TableSchema tableSchema, Map<String, String> tablePropertie
...
+
+            ResultSet pkResultSet = conn.getMetaData().getPrimaryKeys(null, null, tableName);
+            while (pkResultSet.next()) {
+                jdbcPk.add(pkResultSet.getString("COLUMN_NAME"));
+            }
+
             conn.close();
         } catch (SQLException e) {
             throw Status.INTERNAL.withCause(e).asRuntimeException();
         }
```
Comment (chenzl25) [inline java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSinkFactory.java:76]: Let's use `try-with-resources` to improve the code here.
https://stackoverflow.com/questions/8066501/how-should-i-use-try-with-resources-with-jdbc

Comment (wenym1) [reply java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSinkFactory.java]: May do the same for iceberg Hadoop catalog of possible?

**Pull request merged**